### PR TITLE
More deprecations and correspoding cleanups

### PR DESCRIPTION
**Pull request by adamcik**

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -10,8 +10,12 @@ v1.1.0 (UNRELEASED)
 Core API
 --------
 
-- Calling :meth:`mopidy.core.library.LibraryController.search`` with ``kwargs``
-  as the query is no longer supported (PR: :issue:`1090`)
+- Calling the following methods with ``kwargs`` is no longer supported.
```
Comment (jodal) [inline docs/changelog.rst]: Don't mix "deprecated" and "not supported", they might not mean the same to everyone. I read "not supported" as "no longer works".

File: mopidy/core/playlists.py [hunk 2]
```diff
@@ -172,15 +169,21 @@ def filter(self, criteria=None, **kwargs):
 
         .. deprecated:: 1.0
             Use :meth:`as_list` and filter yourself.
+
+        .. deprecated:: 1.1
+            Providing the criteria via ``kwargs`` is no longer supported.
```
Comment (jodal) [inline mopidy/core/playlists.py]: Isn't this very redundant as the entire method is already deprecated?

File: mopidy/core/tracklist.py [hunk 3]
```diff
@@ -398,35 +398,24 @@ def filter(self, criteria=None, **kwargs):
...
-            # ('xyz' or 'abc')
-            filter({'id': [1, 3, 6], 'uri': ['xyz', 'abc']})
-            filter(id=[1, 3, 6], uri=['xyz', 'abc'])
 
         :param criteria: on or more criteria to match by
         :type criteria: dict, of (string, list) pairs
         :rtype: list of :class:`mopidy.models.TlTrack`
+
+        .. deprecated:: 1.1
+            Providing the criteria  via ``kwargs`` is no longer supported.
```
Comment (jodal) [inline mopidy/core/tracklist.py]: s/  via/ via/ (two spaces)

Based on how the deprecated directive is used in other projects, I suggest removing the "is no longer supported" part.

File: mopidy/core/playlists.py (hunk 2 above)
Comment (adamcik) [inline mopidy/core/playlists.py]: Oh yes :-) I can probably just remove this again since I've fixed the tests I wanted to fix by now.

**Comment by jodal** [comment]
Think that was all :-)

**Comment by adamcik** [comment]
Fixed via rebase and squashed fixups.

**Pull request merged**